src/compiler/checker.ts

```diff
@@ -7702,15 +7704,17 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 || isMappedTypeNode(node);
         }
 
-        function getTypeParametersInScope(node: IntroducesNewScopeNode) {
-            return isFunctionLike(node) || isJSDocSignature(node) ? getSignatureFromDeclaration(node).typeParameters : [getDeclaredTypeOfTypeParameter(getSymbolOfDeclaration(node.typeParameter))];
+        function getTypeParametersInScope(node: IntroducesNewScopeNode | ConditionalTypeNode) {

```

Comment:
I kept `ConditionalTypeNode` out of `IntroducesNewScopeNode` because the semantics of it are different. The induction of the scope can't happen automatically like for the others